### PR TITLE
[Docs] Document apiKey usage in remote reindex

### PR DESCRIPTION
**Pull request by nellicus**
This PR documents usage of `ApiKey` through `Headers` in the rest request when calling `_reindex` API.

See https://github.com/elastic/elasticsearch/issues/66211#issuecomment-745247716

### CODE REVIEW

File: docs/reference/docs/reindex.asciidoc [hunk 1]
```diff
@@ -994,6 +995,33 @@ using basic auth or the password will be sent in plain text.
 There are a range of <<reindex-ssl,settings>> available to configure the behaviour of the
  `https` connection.
 
+When using Elastic Cloud, it is also possible to authenticate against the remote cluster
+through the use of a valid Api Key:
```
Comment (debadair) [inline docs/reference/docs/reindex.asciidoc:999]:
```suggestion
through the use of a valid API key:
```

**Review by debadair — APPROVED** [review]
One minor suggestion, but LGTM.

File: docs/reference/docs/reindex.asciidoc (hunk 1 above)
Comment (nellicus) [reply docs/reference/docs/reindex.asciidoc:999]: 👍🏻

**Comment by nellicus** [comment]
Thanks @debadair  I've pushed the change, I see the job is failing https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request+docs-check/5099/console with several errors like:

>09:54:26         java.lang.AssertionError: expected [2xx] status code but api [raw[method=POST path=_reindex]] returned [400 Bad Request] [{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"[remote-elastic-cloud-deployment:9243] not whitelisted in reindex.remote.whitelist","stack_trace":"org.elasticsearch.ElasticsearchException$1: [remote-elastic-cloud-deployment:9243] not whitelisted in reindex.remote.whitelist

>09:54:26 » [2022-04-04T07:50:47,402][ERROR][o.e.x.s.SnapshotRetentionTask] [node-0] error during snapshot retention task
09:54:26 »  org.elasticsearch.repositories.RepositoryMissingException: [my_repository] missing

however I am not familiar with the docs build process and not sure what is to be expected, please let me know if I can take some more steps to get this merged.

**Comment by lockewritesdocs** [comment]
@elasticmachine run docs-check

**Comment by nellicus** [comment]
@elasticmachine run elasticsearch-ci/docs-check

**Comment by lockewritesdocs** [comment]
@elasticmachine update branch

**Comment by lockewritesdocs** [comment]
> Thanks @debadair I've pushed the change, I see the job is failing https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request+docs-check/5099/console with several errors like:
> 
> > 09:54:26         java.lang.AssertionError: expected [2xx] status code but api [raw[method=POST path=_reindex]] returned [400 Bad Request] [{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"[remote-elastic-cloud-deployment:9243] not whitelisted in reindex.remote.whitelist","stack_trace":"org.elasticsearch.ElasticsearchException$1: [remote-elastic-cloud-deployment:9243] not whitelisted in reindex.remote.whitelist
> 
> > 09:54:26 » [2022-04-04T07:50:47,402][ERROR][o.e.x.s.SnapshotRetentionTask] [node-0] error during snapshot retention task
> > 09:54:26 »  org.elasticsearch.repositories.RepositoryMissingException: [my_repository] missing
> 
> however I am not familiar with the docs build process and not sure what is to be expected, please let me know if I can take some more steps to get this merged.

@nellicus, this error indicates that a `[2xx]` response was expected, but a `[400]` response was returned. However, the message doesn't indicate _where_ the error occurred. Looking at the logs, you'll see this message:

```
[ERROR][o.e.s.DocsClientYamlTestSuiteIT] [test] This failing test was generated by documentation starting at reference/docs/reindex.asciidoc:line_1002. It may include many snippets. 
```

If you go to that line in the `reindex.asciidoc` file, the `POST _reindex` request that you added is the one that's generating errors. The indicated host in that request (`https://remote-elastic-cloud-deployment:9243`) doesn't exist, and the `ApiKey` is just a placeholder, so that request will fail every time. I added `// NOTCONSOLE` so that this request isn't tested, which should allow the CI builds to pass.

**Comment by nellicus** [comment]
@lockewritesdocs thank you! I had no idea these strings could be interpreted as commands and be actually run... looks like it's failed again?

**Comment by nellicus** [comment]
@lockewritesdocs I think last failure is because of a comma missing

```json
{
  "source":{
    "remote":{
      "host":"http://${host}", // HERE
      "headers":{
        "Authorization":"ApiKey API_KEY_VALUE"
      }
    },
    "index":"my-index-000001",
    "query":{
      "match":{
        "test":"data"
      }
    }
  },
  "dest":{
    "index":"my-new-index-000001"
  }
}
```

**Comment by nellicus** [comment]
@lockewritesdocs comma after the double quote..

from `"host":"http://${host},"` to `"host":"http://${host}",`

**Comment by lockewritesdocs** [comment]
@nellicus, thanks for your help on this one! Does this apply to previous 8.x versions? If so, I'll apply the appropriate labels and backport accordingly.

**Comment by lockewritesdocs** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.2|[](https://github.com/elastic/elasticsearch/pull/86636)|
|✅|8.1|[](https://github.com/elastic/elasticsearch/pull/86637)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**